enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,502 @@
+
+### Conjunction with Capabilities
+
+In the future, at some degree, [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+approach might help to solve the issue and give a room for supplementing *some* platform-dependent components, however some integral parts of
+OpenShift can not be disabled and still requires a signal about an underlying platform, for example, KCM and MCO, in respect of the enablement of
+[external cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/).
+
+One of the possible examples of the interaction of the capabilities with the "External" platform type would be the MachineAPI and the Machine Api Operator.
+At this point, the MachineAPI has no use in running if there is no machine controller (which is heavily cloud-provider dependent).
+"External" platform type will serve as a signal in the scenario when machine-api capability is enabled and the cluster deployer
+going to satisfy the machine controller requirement. In this case, machine-api-operator would know what to do and deploy
+only generic cloud-independent controllers (such as machine-healthcheck, machineset, and node-link controllers), such
+behaviour will simplify initial cloud-platform enablement and will reduce the necessity of reverse-engineering and replicating a work
+which was already done by Red Hat engineers.
+
+### User Stories
+
+- As a cloud-provider affiliated engineer / platform integrator / RH partner
+  I want to have a mechanism to signal Openshift's built-in operators about additional
+  cloud-provider specific components so that I can inject my own platform-specific controllers into OpenShift
+  to improve the integration between OpenShift and my cloud provider.
+
+  We are aware of examples of supplementing OKD installations with custom machine-api controllers,
+  however, users are experiencing a lot of difficulties on this path due to the necessity of, literally, reverse engineering,
+  manual management of generic MAPI controllers, and so on.
+
+- As a cloud provider whose platform is not integrated into OpenShift,
+  I want to have the Cloud Controller Manager for my infrastructure running in OpenShift from the initial install.
+  Having a platform type that allows for the addition of operators or components which perform platform-specific functionality would help
+  me to create better integrations between OpenShift and my infrastructure platform.
+
+### Goals
+
+- Reduce the necessity to make changes in "openshift/api", "openshift/library-go" and dependant infrastructure-related components during basic cloud-provider integration with OpenShift
+- Make a cloud provider integration process more accessible and simple to external contributors as well as for RH engineers
+- Provide a top-level description of the operation mode for affected components at the first steps
```

Comment:
I had to read this a couple of times before I understood it. If I'm following the intent, what do you think about this revision?

"Define how affected components will operate when the new platform type is selected."

Reply:
we might want to scope this a little tighter and also specify something in the non-goals as well. in the phase 1 part of this enhancement, we want to describe generally what changes we could make to the platform specific operators in phase 3. but we also want to make it clear that before we finalize those changes for phase 3 that we want to have more direct contact with users and partners so that we understand the needs.

so, for the goals, i would expect that we will give a general description of how each component could be modified for better integration in phase 3, **but** we also need to make it clear, possibly via non-goals, that we are not defining the implementations for phase 3 and will need to update this enhancement, or create per-component enhancements, to describe that work when it becomes clearer.

Reply:
I went through goals and non-goals again, actually we have this elaborated more-less:

https://github.com/openshift/enhancements/pull/1234/files#diff-08391a193550547988305aefa40e474c593e97d3b2375245c31015f39f5812b3R149 - mention in non-goals

https://github.com/openshift/enhancements/pull/1234/files#diff-08391a193550547988305aefa40e474c593e97d3b2375245c31015f39f5812b3R173 - note in the "Affected Components" section

Reply:
@lobziik thanks for the links, you are correct we do cover this in detail.

to @mhrivnak 's point, i think we need just need to clean up the language here to make it clearer about our goal for this enhancement. we are describing what we believe, at this time, to be the projected changes that each affected component should take in phase 3.

what about if we rephrase like this?

"Provide an overview of projected changes to affected components that will be planned for a later phase of development."